### PR TITLE
tests: reformat embedded tf manifests

### PR DESCRIPTION
**Pull request by mhoffm-aiven**
* reformat embedded terraform manifests
* small tool to auto format embedded terraform manifests

### CODE REVIEW

File: hack/testmanifestfmt/go.mod [hunk 1]
```diff
@@ -0,0 +1,19 @@
+module github.com/aiven/terraform-provider-aiven/hack/testmanifestfmt
```
Comment (nkchern-avn) [inline hack/testmanifestfmt/go.mod:1]: dir name `hack` looks strange. Am I missing any context here?

File: hack/testmanifestfmt/main.go [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+	defer func() { _ = os.RemoveAll(tmpdir) }()
+
+	tf, err := tfexec.NewTerraform(tmpdir, execPath)
+	if err != nil {
+		log.Fatalf("[ERROR] unable to create terraform handle: %s", err)
+	}
+
+	log.Println("[INFO] installed terraform, formatting files")
+
+	for i := range flag.Args() {
```
Comment (nkchern-avn) [inline hack/testmanifestfmt/main.go:50]: `for _, path := range ...` would save you one line.

File: hack/testmanifestfmt/main.go [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+}
+
+func main() {
+	ctx := context.Background()
+
+	log.Println("[INFO] installing terraform")
+
+	installer := &releases.ExactVersion{
+		Product: product.Terraform,
+		Version: version.Must(version.NewVersion("1.0.6")),
```
Comment (nkchern-avn) [inline hack/testmanifestfmt/main.go:32]: hardcoded version, to constants at least?

File: hack/testmanifestfmt/main.go [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+		if err = reformatFileAtPath(ctx, path, tf); err != nil {
+			log.Println("[ERROR]: unable to format ", path, ": ", err)
+		}
+	}
+
+	log.Println("[INFO] formatted all files")
+}
+
+func reformatFileAtPath(ctx context.Context, path string, tf *tfexec.Terraform) error {
+	file, err := os.OpenFile(path, os.O_RDWR, 0755)
```
Comment (nkchern-avn) [inline hack/testmanifestfmt/main.go:63]: magic number 0755

File: hack/testmanifestfmt/main.go [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+	}
+
+	log.Println("[INFO] installed terraform, formatting files")
+
+	for i := range flag.Args() {
+		path := flag.Args()[i]
+
+		log.Println("[INFO] formatting ", path)
+		if err = reformatFileAtPath(ctx, path, tf); err != nil {
+			log.Println("[ERROR]: unable to format ", path, ": ", err)
```
Comment (nkchern-avn) [inline hack/testmanifestfmt/main.go:55]: `log.Formatf`?

File: hack/testmanifestfmt/main.go [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+	if _, err := file.Seek(0, 0); err != nil {
+		return fmt.Errorf("unable to seek beginning of file at path '%s': %w", path, err)
+	}
+
+	format.Node(file, fset, parsed)
+
+	return nil
+}
+
+func isTerraformManifestMaybe(s string) bool {
```
Comment (nkchern-avn) [inline hack/testmanifestfmt/main.go:117]: I'd argue that `Maybe` is redundant here. `isTerraformManifest` clearly explains what the function is about.

File: hack/testmanifestfmt/main.go [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+		}
+		// TODO: better way to check if the string is a terraform manifest
+		if !isTerraformManifestMaybe(lit.Value) {
+			return false
+		}
+
+		unquoted := strings.Trim(lit.Value, "`")
+
+		if formatted, err := tf.FormatString(ctx, unquoted); err != nil {
+			log.Println("[ERROR]: unable to format node ", unquoted, ": ", err)
```
Comment (nkchern-avn) [inline hack/testmanifestfmt/main.go:93]: same, `log.Printf`?

File: hack/testmanifestfmt/main.go [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+			lit.Value = "`" + formatted + "`"
+		}
+		return false
+	})
+
+	if hadFmtErr {
+		return fmt.Errorf("unable to format file at path '%s'", path)
+	}
+
+	if err := file.Truncate(0); err != nil {
```
Comment (nkchern-avn) [inline hack/testmanifestfmt/main.go:105]: would it be cleaner and more straightforward to read the file(`os.Open`) above:
https://github.com/aiven/terraform-provider-aiven/pull/649/files#diff-63a02b7acc7a67a3b6782cf1d89ddd8597c120ff82c5196625f67dc6284788baR63

and then overwrite this file here using `os.Create`
This way you can get rid this Truncate/Seek dance.

**Review by nkchern-avn — COMMENTED** [review]
reviewed. All comments are fairly minor.

File: hack/testmanifestfmt/main.go (hunk 2 above)
Comment (mhoffm-aiven) [reply hack/testmanifestfmt/main.go:50]: done

File: hack/testmanifestfmt/main.go (hunk 3 above)
Comment (mhoffm-aiven) [reply hack/testmanifestfmt/main.go:32]: done

File: hack/testmanifestfmt/main.go (hunk 4 above)
Comment (mhoffm-aiven) [reply hack/testmanifestfmt/main.go:63]: i think thats fair as a file mode though

File: hack/testmanifestfmt/main.go (hunk 6 above)
Comment (mhoffm-aiven) [reply hack/testmanifestfmt/main.go:117]: done

File: hack/testmanifestfmt/main.go [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+		return fmt.Errorf("unable to truncate file at path '%s': %w", path, err)
+	}
+	if _, err := file.Seek(0, 0); err != nil {
+		return fmt.Errorf("unable to seek beginning of file at path '%s': %w", path, err)
+	}
+
+	format.Node(file, fset, parsed)
```
[the remark this answers is not in the document]
Comment (mhoffm-aiven) [reply hack/testmanifestfmt/main.go:112]: done

File: hack/testmanifestfmt/main.go (hunk 5 above)
Comment (mhoffm-aiven) [reply hack/testmanifestfmt/main.go:57]: as discussed offline, ill dismiss the logging comments

File: hack/testmanifestfmt/main.go (hunk 7 above)
Comment (mhoffm-aiven) [reply hack/testmanifestfmt/main.go:95]: as discussed offline, ill dismiss the logging comments

File: hack/testmanifestfmt/main.go (hunk 8 above)
Comment (mhoffm-aiven) [reply hack/testmanifestfmt/main.go:107]: since it works with the truncate and seek ill keep it for now

**Review by ivan-savciuc — APPROVED** [review]
LGTM

**Review by ivan-savciuc — APPROVED** [review]

**Pull request merged**